src/client/src/components/PublicUserProfile.jsx

```diff
@@ -11,18 +11,21 @@ const PublicUserProfile = () => {
   let locationURL = useLocation().pathname;
   let profileId = useLocation().pathname.split("/")[3];
 
-  //HTTP request to backend to fetch user info
+  //HTTP Request in Backend to fetch user info
+  const fetchProfile = async () => {
+    try {
+      const response = await axios.get(
+        `http://localhost:9000/users/profile/${profileId}`
+      );
+      setPublicUser(response.data);
+      console.log(response.data);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  //Make the request once and only when locationURL changes
   useEffect(() => {
-    const fetchProfile = async () => {
-      try {
-        const response = await axios.get(
-          `http://localhost:9000/users/profile/${profileId}`
-        );
-        setPublicUser(response.data);
-      } catch (error) {
-        console.error(error);
-      }
-    };
     fetchProfile();
   }, [locationURL]);
```

Comment:
So the component re-renders when the URL changes?

Reply:
yes ? are you suggesting that it should depend on something else?

Reply:
No this makes sense, just wanted to confirm!